src/Symfony/Component/HttpFoundation/Cookie.php

```diff
@@ -62,7 +62,15 @@ public static function fromString($cookie, $decode = false)
         $part = array_shift($parts);
 
         $name = $decode ? urldecode($part[0]) : $part[0];
-        $value = isset($part[1]) ? ($decode ? urldecode($part[1]) : $part[1]) : null;
+        $value = null;
+
+        if (isset($part[1])) {
```

Comment:
What if the input is `foo==`? are we sure that `$part[1]` is an empty string? Or HeaderUtils replaced empty strings by null? 
Maybe we can add  test for this case?

Reply:
Currently, that example would be parsed to have `foo` as the cookie value. I am actually not sure what would be the desired value here though.

Reply:
I think I found a solution.